config.py

```diff
@@ -148,8 +148,8 @@ def get_cloud_init(name, base, password, provider, privateData, slavePortnum):
             )
             c['slaves'].append(slave)
     else:
-        for i, password in enumerate(slaveConfig['passwords']):
-            name = '%s-%d' % (base, i)
+        for index, password in enumerate(slaveConfig['passwords']):
```

Comment:
Not dynamically. They need to be communicated to the slave out-of-band, and thus a record needs to be kept of them somewhere. Certainly they should be originally created randomly.